lib/std/os.zig

```diff
@@ -254,6 +254,101 @@ pub fn close(fd: fd_t) void {
     }
 }
 
+pub const FChmodError = error{
+    AccessDenied,
+    InvalidFileDescriptor,
+    InputOutput,
+    SymLinkLoop,
+    FileNotFound,
+    SystemResources,
+    ReadOnlyFileSystem,
+} || UnexpectedError;
+
+/// Changes the mode of the file referred to by the file descriptor.
+/// The process must have the correct privileges in order to do this
+/// successfully, or must have the effective user ID matching the owner
+/// of the file.
+pub fn fchmod(fd: fd_t, mode: mode_t) FChmodError!void {
+    while (true) {
+        const res = if (builtin.link_libc) blk: {
+            const rc = std.c.fchmod(fd, mode);
+            break :blk std.c.getErrno(rc);
+        } else if (builtin.os.tag == .linux) blk: {
+            const rc = linux.fchmod(fd, mode);
+            break :blk linux.getErrno(rc);
+        } else blk: {
+            const rc = system.fchmod(fd, mode);
+            break :blk system.getErrno(rc);
+        };
+
+        switch (res) {
+            .SUCCESS => return,
+            .INTR => continue,
+
+            .FAULT => unreachable,
+            .INVAL => unreachable,
+            .ACCES => return error.AccessDenied,
+            .BADF => return error.InvalidFileDescriptor,
```

Comment:
I believe this should be `unreachable` and same thing with fchmod.

Reply:
Which should be unreachable? Access denied should surely be an error to be handled, and I've seen other code in `os.zig` use a similar pattern for `BADF`.

Reply:
Talking about `BADF`.  Almost everything uses unreachable.  There may be times where it's not used and I think those are times when `BADF` is overloaded to mean multiple things (i.e. not just a bad file descriptor).

Reply:
Let's talk about `BADF` in just one conversation.